name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java

```diff
@@ -55,31 +76,24 @@ public String getTermCurrency()
         return termCurrency;
     }
 
-    @Override
-    public Money convert(LocalDate date, Money amount)
-    {
-        try
-        {
-            return CurrencyConverter.super.convert(date, amount);
-        }
-        catch (MonetaryException e)
-        {
-            // testing: any other currency will be converted 1:1
-            if (!amount.getCurrencyCode().equals(series.getBaseCurrency()))
-                return Money.of(termCurrency, amount.getAmount());
-            else
-                throw e;
-        }
-    }
-
     @Override
     public ExchangeRate getRate(LocalDate date, String currencyCode)
     {
         if (termCurrency.equals(currencyCode))
             return new ExchangeRate(date, BigDecimal.ONE);
 
-        if (!currencyCode.equals(series.getBaseCurrency()))
-            throw new MonetaryException();
+        ExchangeRateTimeSeries series;
+        if (currencyCode.equals("USD") && termCurrency.equals("EUR"))
+            series = USD_EUR;
+        else if (currencyCode.equals("EUR") && termCurrency.equals("USD"))
+            series = EUR_USD;
+        else if (currencyCode.equals("CHF") && termCurrency.equals("EUR"))
+            series = CHF_EUR;
+        else if (currencyCode.equals("EUR") && termCurrency.equals("CHF"))
+            series = EUR_CHF;
+        else
+            // testing: any other currency will be converted 1:1
+            return new ExchangeRate(date, BigDecimal.ONE);
```

Comment:
… and this is where, in the new code, an exchange rate of 1 is passed to the superclass’s `convert()`, which uses it to do a 1:1 conversion.